### ISSUE TITLE
Feature Request: Implement IDN Method for Instrument Identification

### ISSUE THREAD

**Issue by pgwijesinghe** [levylabpitt/Instrument-Framework#109]
I propose adding a feature where all instruments output an identification (IDN) dictionary upon request. This feature would be similar to outputting the set of available commands upon a 'HELP' request.

e.g.
```
[In] {"jsonrpc": "2.0", 
            "method": "IDN",
            "id": "id"} 
[Out] {"jsonrpc": "2.0", 
            "result": {'vendor': 'Levylab',
                          'model': 'MC Lockin',
                          'serial': None,
                          'firmware': 'v2.15.4.4'},
            "id": "id"}
```

**Comment by ciozi137** [comment]
Good idea. Any ideas how to generate a serial number for a soft instrument? Things like the cryostats are more or less straightforward because they could be derived from the hardware. But we have X lockins that are all identical... maybe I can get a number from the pxi chassis/4461 cards

**Comment by ciozi137** [comment]
![image](https://github.com/levylabpitt/Instrument-Framework/assets/26659428/4fce26fb-fbc0-44b8-9f9b-bb15de1fb27c)

`Write IDN` is a public method that the child class will call (where? `onProcessStarted`?) to set the serial number and firmware number

**Comment by ciozi137** [comment]
![image](https://github.com/levylabpitt/Instrument-Framework/assets/26659428/6c8d18da-5d33-449f-9192-861de64f8037)

**Comment by ciozi137** [comment]
Protected method to set IDN data:
![image](https://github.com/levylabpitt/Instrument-Framework/assets/26659428/749c8dd8-00e0-4c6b-bc0a-4bf05110699c)

**Comment by ciozi137** [comment]
![image](https://github.com/levylabpitt/Instrument-Framework/assets/26659428/a9da7ec6-2476-400d-a310-f299875c9f01)

IDN Data Flow:
1. `Child Instrument:process.vi` calls `Open Hardware.vi`
2. `Open Hardware.vi` opens connection to the hardware, and then calls `Write IDN.vi`
3. `Write IDN.vi` queries the S/N (or reads a constant, it's up to the developer) and then calls `Instrument.sendMessageToProcess.vi`, the message is then handled by `Handle Command.vi` in the `Instrument:process.vi`.
